src/Ucommerce.Avenue.Umbraco/Controllers/CategoryController.cs

```diff
@@ -68,12 +63,13 @@ private IList<ProductViewModel> MapProductsInCategories(Category category)
             IList<Facet> facetsForQuerying = System.Web.HttpContext.Current.Request.QueryString.ToFacets();
             var productsInCategory = new List<ProductViewModel>();
 
-            foreach (var subcategory in category.Categories)
+            foreach (var subCategoryGuid in category.Categories)
             {
-                productsInCategory.AddRange(MapProductsInCategories(subcategory));
+                var subCategory = CatalogLibrary.GetCategory(subCategoryGuid);
```

Comment:
There's a _SELECT N+1_ here. Not going to perform well.
Better option would be to get all sub cats in one query, to a `.ToLookup` and then index that lookup inside the loop.

The fact that it's recursive makes it hard to make it perform well. For the category navigation component, @mufoxe and I agreed that for now, we'd support two levels and not make it recursive. Later, we'll add js code to load more levels. I'm not sure if the same logic is applicable here, though?

Reply:
you are right.
I was too focused at replicating all 1:1, so I didn't think to use what the new structure to our advantage

Reply:
There's still a SELECT N+1
If you have a query inside a loop, that's one.
All querying should be done before the `foreach`.